build-config/scripts/make-devices.pl

```diff
@@ -179,6 +179,12 @@ sub devices
 	makedev("mmcblk0p${i}", "b", 179, ${i}, ${root}, ${root}, 0660);
     }
 
+    print "NVMe SSD access: nvme0n1 nvme0n1p{1..7}\n";
+    makedev("nvme0n1", "b", 259, 0, ${root}, ${root}, 0660);
+    foreach my $i (1..7) {
+    makedev("nvme0n1p${i}", "b", 259, ${i}, ${root}, ${root}, 0660);
+    }
```

Comment:
I do not think creating these device nodes at compile time is going to work in general.
The major number used here, 259, is not necessarily the number the Linux kernel will choose when the nvme driver is registered.  259 might be the correct number on your system, but in general it will not be correct.

I have two thoughts one this:

1. It might be time to move away from the static dev names.  Use a helper program, like `mdev`, to populate /dev/ dynamically, similar to how udev works.

2. use a hybrid approach, like how the VIRT_IO disk devices are create for the virtual machine.   These are the `/dev/vd[a-z]` devices created in /etc/init.d/makedev.sh. https://github.com/opencomputeproject/onie/blob/master/rootconf/default/etc/init.d/makedev.sh#L93-L117

Can you try using `mdev` and see if helps here.  Here's the help on `mdev`:
```
ONIE:/ # mdev --help
BusyBox v1.25.1 (2017-06-22 15:39:41 PDT) multi-call binary.

Usage: mdev [-s]

mdev -s is to be run during boot to scan /sys and populate /dev.

Bare mdev is a kernel hotplug helper. To activate it:
        echo /sbin/mdev >/proc/sys/kernel/hotplug

It uses /etc/mdev.conf with lines
        [-][ENV=regex;]...DEVNAME UID:GID PERM [>|=PATH]|[!] [@|$|*PROG]
where DEVNAME is device name regex, @major,minor[-minor2], or
environment variable regex. A common use of the latter is
to load modules for hotplugged devices:
        $MODALIAS=.* 0:0 660 @modprobe "$MODALIAS"

If /dev/mdev.seq file exists, mdev will wait for its value
to match $SEQNUM variable. This prevents plug/unplug races.
To activate this feature, create empty /dev/mdev.seq at boot.

If /dev/mdev.log file exists, debug log will be appended to it.
```

I do not think we need to use the config file.  I think we just need to activate it.

The mdev helper could be activated in /etc/init.d/makedev.sh.

You could test it by first activating mdev, then create a couple partitions using `sgdisk`.  Then check if the device nodes were created in /dev/.  Something like this:
```
# echo /sbin/mdev >/proc/sys/kernel/hotplug
# sgdisk -n 4:: /dev/nvme0n1
# ls -l /dev/nvme0n1*
```
If that is working we should consider activating mdev for hotplug by default.